### PR TITLE
Create tunnel command

### PR DESCRIPTION
**Pull request by ostcar**
The default should be to open tunnels to all services

### CODE REVIEW

File: pkg/manage/cmd_tunnel.go [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+	"github.com/OpenSlides/openslides-manage-service/proto"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc/metadata"
+)
+
+const helpTunnel = `Opens local ports to all services and creates
+tunnels into the OpenSlides network to the services they belong to.
+
+Without any argument, the command creates tunnels to all services at there
```
Comment (normanjaeckel) [inline pkg/manage/cmd_tunnel.go:20]: Typo `their`

File: pkg/manage/cmd_tunnel.go [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc/metadata"
+)
+
+const helpTunnel = `Opens local ports to all services and creates
+tunnels into the OpenSlides network to the services they belong to.
+
+Without any argument, the command creates tunnels to all services at there 
+default ports. To specify tunnels, the flag "-L" can be used or the name
+of a known services as argument.
```
Comment (normanjaeckel) [inline pkg/manage/cmd_tunnel.go:22]: Typo`service`

File: pkg/manage/cmd_tunnel.go [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+		}
+
+		wg.Wait()
+
+		return nil
+	}
+	return &cmd
+}
+
+func tunnelParseArgument(args []string) map[string]string {
```
Comment (normanjaeckel) [inline pkg/manage/cmd_tunnel.go:119]: Add docstring and maybe describe with an example with map came back here.

File: pkg/manage/cmd_tunnel.go [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+//
+// Listens on the given localAddr, sends all data via grpc to the manage server
+// and there redirect it to the remoteAddr.
+//
+// Blocks until the tunnel is closed.
+func newTunnel(service proto.ManageClient, localAddr string, remoteAddr string) error {
+	// Listen on localAddr
+	lst, err := net.Listen("tcp", localAddr)
+	if err != nil {
+		return fmt.Errorf("start listening on %s: %v", localAddr, err)
```
Comment (normanjaeckel) [inline pkg/manage/cmd_tunnel.go:141]: Why not %w instead of %v?

File: pkg/manage/cmd_tunnel.go [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+		for {
+			bs, err := rw.Read(buff)
+			if err != nil {
+				if err != io.EOF {
+					fromRW <- fmt.Errorf("receiving data: %w", err)
+				}
+				return
+			}
+
+			err = sr.Send(&proto.TunnelData{
```
Comment (normanjaeckel) [inline pkg/manage/cmd_tunnel.go:248]: Here you may choose the common style `if err := ...; err != nil {...}`

File: pkg/manage/cmd_tunnel.go [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+// SendReceiver reads and writes from a grpc tunnel connection.
+type SendReceiver interface {
+	Recv() (*proto.TunnelData, error)
+	Send(*proto.TunnelData) error
+}
+
+// copyStream connects the grcp connection with a io.ReadWriter.
+//
+// Blocks until one connection is closed.
+func copyStream(sr SendReceiver, rw io.ReadWriter) error {
```
Comment (normanjaeckel) [inline pkg/manage/cmd_tunnel.go:210]: Maybe split this function up into three. One "main" copy function and two functions for the goroutines.

**Review by normanjaeckel — APPROVED** [review]

File: pkg/manage/cmd_tunnel.go (hunk 1 above)
Comment (ostcar) [reply pkg/manage/cmd_tunnel.go:20]: done

File: pkg/manage/cmd_tunnel.go (hunk 3 above)
Comment (ostcar) [reply pkg/manage/cmd_tunnel.go:119]: done

File: pkg/manage/cmd_tunnel.go (hunk 4 above)
Comment (ostcar) [reply pkg/manage/cmd_tunnel.go:141]: done

File: pkg/manage/cmd_tunnel.go (hunk 5 above)
Comment (ostcar) [reply pkg/manage/cmd_tunnel.go:248]: done

File: pkg/manage/cmd_tunnel.go (hunk 6 above)
Comment (ostcar) [reply pkg/manage/cmd_tunnel.go:210]: This was a good idea. Done.

File: pkg/manage/cmd_tunnel.go (hunk 2 above)
Comment (ostcar) [reply pkg/manage/cmd_tunnel.go]: done

**Pull request merged**